### DISCUSSION TITLE
Creating relational dataset from sequential data

### DISCUSSION THREAD

**Discussion by yeongwook** [LukasZahradnik/PyNeuraLogic#53] [Q&A]
Hi,

We have a sequential dataset and would like to transform it into a relational form so we can use the MolecularGNN example of the PyNeuraLogic framework (https://github.com/LukasZahradnik/PyNeuraLogic/blob/master/examples/MolecularGNN.ipynb). We only want to use these columns of our attached dataset (user_id, skill_id, correct).

Thank you

[skill_builder_data2.csv](https://github.com/LukasZahradnik/PyNeuraLogic/files/11366449/skill_builder_data2.csv)

**Comment by LukasZahradnik** [comment]
Hello,

PyNeuraLogic has a CVS dataset that can convert CSV files to relations. Now, assuming that "user_id" + "correct" forms labels and "user_id" + "skill_id" input examples, the dataset could be converted to relations in the following ways:

```py
from neuralogic.dataset import CSVDataset, CSVFile, FileDataset

path = "./skill_builder_data2.csv"

queries = CSVFile("predict", path, value_column="correct", term_columns=["user_id"], header=True, n_rows=540)

examples = [
    CSVFile("feature", path, value_column="correct", term_columns=["user_id", "skill_id"], header=True, n_rows=540),
]

dataset = CSVDataset(examples, queries)

# 1. to_dataset()
dataset = dataset.to_dataset()

# 2. Or to file dataset
dataset = dataset.to_dataset().dump_to_file("./queries.txt", "./examples.txt")
dataset = FileDataset("./examples.txt", "./queries.txt")
```

I don't know your exact use case, so you might need to tweak some things - e.g., relation names ("predict" and "feature"). The example takes only the first 540 rows; the rest seemed to be empty.

You might also need to convert the CSV to utf-8.
